server/src/main/java/org/apache/druid/server/http/SelfDiscoveryResource.java

```diff
@@ -44,58 +47,68 @@
  * DI configuration phase.
  */
 @Singleton
+@Path("/status/selfDiscovered")
 public class SelfDiscoveryResource
 {
-  private BooleanSupplier selfDiscovered;
+  private final List<BooleanSupplier> selfDiscoveredRoles;
 
   @Inject
   public SelfDiscoveryResource(
       @Self DruidNode thisDruidNode,
-      @Self NodeRole thisNodeRole,
+      @Self Set<NodeRole> thisNodeRoles,
       DruidNodeDiscoveryProvider nodeDiscoveryProvider,
       Lifecycle lifecycle
   )
   {
-    Lifecycle.Handler selfDiscoveryListenerRegistrator = new Lifecycle.Handler()
-    {
-      @Override
-      public void start()
-      {
-        selfDiscovered = nodeDiscoveryProvider.getForNode(thisDruidNode, thisNodeRole);
-      }
+    selfDiscoveredRoles = Lists.newArrayListWithExpectedSize(thisNodeRoles.size());
+    thisNodeRoles.forEach(
+        thisNodeRole -> {
+          Lifecycle.Handler selfDiscoveryListenerRegistrator = new Lifecycle.Handler()
+          {
+            @Override
+            public void start()
+            {
+              selfDiscoveredRoles.add(nodeDiscoveryProvider.getForNode(thisDruidNode, thisNodeRole));
+            }
 
-      @Override
-      public void stop()
-      {
-        // do nothing
-      }
-    };
-    // Using Lifecycle.Stage.SERVER because DruidNodeDiscoveryProvider should be already started when
-    // selfDiscoveryListenerRegistrator.start() is called.
-    lifecycle.addHandler(selfDiscoveryListenerRegistrator, Lifecycle.Stage.SERVER);
+            @Override
+            public void stop()
+            {
+              // do nothing
+            }
+          };
+          // Using Lifecycle.Stage.SERVER because DruidNodeDiscoveryProvider should be already started when
+          // selfDiscoveryListenerRegistrator.start() is called.
+          lifecycle.addHandler(selfDiscoveryListenerRegistrator, Lifecycle.Stage.SERVER);
+        }
+    );
   }
 
   /** See the description of this endpoint in api-reference.md. */
   @GET
-  @Path("/status/selfDiscoveredStatus")
+  @Path("/status")
```

Comment:
Since this changes the API path of `/status/selfDiscoveredStatus` to `/status/selfDiscovered/status` the docs in api-reference.md need updated

Reply:
Oops, thanks. Fixed.